server/data/reviews.js

```diff
@@ -166,18 +174,18 @@ const exportedMethods = {
     );
 
     if (updatedInfo.modifiedCount === 0) {
-      throw { message: "Update Failed", code: 500 };
+      throw "could not update successfully";
     }
 
     const rev = await this.getReviewById(id);
     let productId = rev.product;
     let rateall = 0;
     const prodreview = await this.getReviewbyProductId(productId);
     console.log(prodreview);
-    for (let i = 0; i < prodreview.length; i++) {
-      rateall = parseInt(prodreview[i].rating) + rateall;
+    let averagerate = 0;
+    if (prodreview.length != 0) {
+      averagerate = rateall / prodreview.length;
```

Comment:
not sure if this is on purpose. Can we double check this ?

Reply:
This is the fix "error when deleting the last review for a product, rating would become NaN